keg_auth/libs/decorators.py

```diff
@@ -55,30 +55,52 @@ def decorate_class(self, cls):
         # when decorating a view class, all of the class's route methods will submit to the given
         #   auth. The view may already have check_auth defined, though, so make sure we still call
         #   it.
-        old_check_auth = getattr(cls, 'check_auth', lambda: None)
-
-        def new_check_auth(*args, **kwargs):
-            self.check_auth(instance=args[0])
-
-            # the original check_auth method on the view may take any number of args/kwargs. Use
-            #   logic similar to keg.web's _call_with_expected_args, except that method does not
-            #   fit this case for bound methods
-            try:
-                # validate_arguments is made for a function, not a class method
-                # so we need to "trick" it by sending self here, but then
-                # removing it before the bound method is called below
-                pass_args, pass_kwargs = validate_arguments(old_check_auth, args, kwargs.copy())
-            except ArgumentValidationError as e:
-                msg = _('Argument mismatch occured: method=%s, missing=%s, '
-                        'extra_keys=%s, extra_pos=%s.'
-                        '  Arguments available: %s') % (old_check_auth, e.missing,
-                                                        e.extra, e.extra_positional,
-                                                        kwargs)  # pragma: no cover
-                raise ViewArgumentError(msg)  # pragma: no cover
-
-            return old_check_auth(*pass_args, **pass_kwargs)
-
-        cls.check_auth = new_check_auth
+        old_check_auth = getattr(cls, 'check_auth', None)

```

Comment:
This would be better if
* reduce code duplication. These two blocks are doing the same thing, just with two different method names
* raise an exception if neither name fits, because in that case, this decorator should not be used